egs/heroico/s5/run.sh

```diff
@@ -0,0 +1,451 @@
+	| \
+	sed -e s/
// \
+	| \
+	local/heroico_answers_make_lists.pl
+
+    utils/fix_data_dir.sh \
+	$tmpdir/heroico/answers
+
+    cat \
+	$recordings_transcripts \
+	| \
+	iconv -f ISO-8859-1 -t UTF-8 \
+	| \
+	sed -e s/
// \
+	    | \
+	local/heroico_recordings_make_lists.pl
+
+    utils/fix_data_dir.sh \
+	$tmpdir/heroico/recordings
+
+    # consolidate heroico lists
+    mkdir -p $tmpdir/heroico/lists
+
+    for x in wav.scp utt2spk text; do
+	cat \
+	    $tmpdir/heroico/answers/$x \
+	    $tmpdir/heroico/recordings/$x \
+	    | \
+	    sed -e s/
// \
+		| \
+	    sort \
+		-k1,1 \
+		-u \
+		> \
+		$tmpdir/heroico/lists/$x
+    done
+
+    utils/fix_data_dir.sh \
+	$tmpdir/heroico/lists
+fi
+
+if [ $stage -le 1 ]; then
+    #  make separate lists for usma native and nonnative
+    cat \
+	$usma_transcripts \
+	| \
+	iconv -f ISO-8859-1 -t UTF-8 \
+	| \
+	sed -e s/
// \
+	    | \
+	    local/usma_native_make_lists.pl
+
+    cat \
+	$usma_transcripts \
+	| \
+	iconv -f ISO-8859-1 -t UTF-8 \
+	| \
+	sed -e s/
// \
+	    | \
+	local/usma_nonnative_make_lists.pl
+
+    for n in native nonnative; do
+	mkdir -p $tmpdir/usma/$n/lists
+
+
+	for x in wav.scp utt2spk text; do
+	    sort \
+		$tmpdir/usma/$n/$x \
+		> \
+		$tmpdir/usma/$n/lists/$x
+	done
+
+	utils/fix_data_dir.sh \
+	    $tmpdir/usma/$n/lists
+    done
+
+    mkdir -p data/train
+    mkdir -p $tmpdir/lists
+
+    # get training lists
+    for x in wav.scp utt2spk text; do
+	cat \
+	    $tmpdir/heroico/answers/${x} \
+	    $tmpdir/heroico/recordings/${x} \
+	    | \
+	    sed -e s/
// \
+	    > \
+	    $tmpdir/lists/$x
+
+	sort \
+	    $tmpdir/lists/$x \
+	    > \
+	    data/train/$x
+    done
+
+    utils/utt2spk_to_spk2utt.pl \
+	data/train/utt2spk \
+	| \
+	sort \
+	    > \
+	    data/train/spk2utt
+
+    utils/fix_data_dir.sh \
+	data/train
+
+# make testing  lists
+
+    mkdir \
+	-p \
+	data/test \
+	data/native \
+	data/nonnative \
+	$tmpdir/usma/lists
+
+    for x in wav.scp text utt2spk; do
+	# get testing lists
+	for n in native nonnative; do
+	    cat \
+		$tmpdir/usma/$n/lists/$x \
+		>> \
+		$tmpdir/usma/lists/$x
+	done
+
+	sort \
+	    $tmpdir/usma/lists/$x \
+	    > \
+	    data/test/$x
+
+	for n in native nonnative; do
+	    sort \
+		$tmpdir/usma/$n/$x \
+		> \
+		data/$n/$x
+	done
+    done
+
+    # spk2utt
+    for n in native nonnative test; do
+	utils/utt2spk_to_spk2utt.pl \
+	    data/$n/utt2spk \
+	    | \
+	    sort \
+		> \
+		data/$n/spk2utt
+
+	utils/fix_data_dir.sh \
+	    data/$n
+    done
+fi
+
+if [ $stage -le 2 ]; then
+    # prepare a dictionary
+    mkdir -p data/local/dict
+
+    local/prepare_dict.sh
+
+    # prepare the lang directory
+    utils/prepare_lang.sh \
+	data/local/dict \
+	"<UNK>" \
+	data/local/lang \
+	data/lang   || exit 1;
+fi
+
+if [ $stage -le 3 ]; then
+    # prepare lm on training transcripts
+    local/prepare_lm.sh
+
+    # make lm fst
+    mkdir -p data/local/lm
+
+    gunzip language_models/lm_threegram.arpa.gz 
```

Comment:
there is a script utils/format_lm.sh that handles the whole conversion process -- unless you need something very specific, I think it would be better to rely on that script.

Reply:
I mean conversion ARPA->G.fst